databricks_cli/pipelines/cli.py

```diff
@@ -180,9 +215,24 @@ def _get_pipeline_id(spec_arg, spec, pipeline_id):
     if bool(spec_arg) or bool(spec):
         src = spec_arg if bool(spec_arg) else spec
         pipeline_id = _read_spec(src)["id"]
+    msg = _validate_pipeline_id(pipeline_id)
+    if msg is not None:
+        error_and_quit(msg)
     return pipeline_id
 
 
+def _validate_pipeline_id(pipeline_id):
+    """
+    Checks if the pipeline_id only contain -, _ and alphanumeric characters
+    """
+    if len(pipeline_id) == 0:
+        return 'Empty pipeline id provided'
+    if not set(pipeline_id) <= PIPELINE_ID_PERMITTED_CHARACTERS:
+        message = 'Pipeline id {} has invalid character(s)\n'.format(pipeline_id)
+        message += "Valid characters are: _ - a-z A-Z 0-9"
+        return message
```

Comment:
Having validate call error_and_quit makes the code untestable. Click tries to capture the actual error and drops it to raise a Click error for having a context during testing.

Reply:
That makes this API a lot more awkward to use. Can the tests set up a context so the real error gets surfaced?

Reply:
I spent some time trying to do the same, I'll have a look again